hypershift/hosted-wrapper.py

```diff
@@ -0,0 +1,592 @@
+            tags_list.append(tag['ref'].split('/')[-1].split('v')[-1])
+        last_version = sorted(tags_list,key=version.StrictVersion)[-1]
+        logging.info('Identified latest release as %s' % last_version)
+        url = 'https://github.com/openshift-online/ocm-cli/releases/download/v' + last_version + '/ocm-linux-amd64'
+        with urllib.request.urlopen(url) as response, open(my_path + '/ocm', 'wb') as out_file:
+            shutil.copyfileobj(response, out_file)
+        os.chmod(my_path + '/ocm', 0o777)
+        ocm_cmnd = my_path + "/ocm"
+    logging.info('Testing ocm command with: ocm -h')
+    ocm_cmd = [ocm_cmnd, "-h"]
+    ocm_process = subprocess.Popen(ocm_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    ocm_stdout,ocm_stderr = ocm_process.communicate()
+    if ocm_process.returncode != 0:
+        logging.error('%s unable to execute -h' % ocm_cmnd)
+        logging.error(ocm_stderr.strip().decode("utf-8"))
+        return 1
+    logging.info('ocm command validated with -h. Directory is %s' % my_path)
+    if hypershift_cmnd is None:
+        logging.info('hypershift command not provided')
+        logging.info('Cloning hypershift repo and compiling cli from https://github.com/openshift/hypershift/')
+        Repo.clone_from("https://github.com/openshift/hypershift.git", my_path + '/hypershift')
+        os.chdir(my_path + '/hypershift')
+        logging.info('Compiling hypershift cli on %s' % my_path + '/hypershift')
+        make_cmd = ["make", "build"]
+        make_process = subprocess.Popen(make_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        make_stdout,make_stderr = make_process.communicate()
+        if make_process.returncode != 0:
+            logging.error('%s unable to execute' % make_cmd)
+            logging.error(make_stderr.strip().decode("utf-8"))
+            return 1
+        logging.info('hypershift cli compiled and stored at %s' % my_path + 'hypershift/bin/hypershift')
+        os.chmod(my_path + '/hypershift/bin/hypershift', 0o777)
+        hypershift_cmnd = my_path + "/hypershift/bin/hypershift"
+    logging.info('Testing hypershift command with: hypershift -h')
+    hypershift_cmd = [hypershift_cmnd, "-h"]
+    hypershift_process = subprocess.Popen(hypershift_cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    hypershift_stdout,hypershift_stderr = hypershift_process.communicate()
+    if hypershift_process.returncode != 0:
+        logging.error('%s unable to execute -h' % hypershift_cmnd)
+        logging.error(hypershift_stderr.strip().decode("utf-8"))
+        return 1
+    logging.info('hypershift command validated with -h. Directory is %s' % my_path)
+    return (ocm_cmnd, hypershift_cmnd)
+
+def _get_mgmt_cluster_info(ocm_cmnd, mgmt_cluster,es,index,index_retry,uuid,hostedclusters,hosted_workers):
+    logging.info('Getting Management Cluster Information from %s' % mgmt_cluster)
+    ocm_command = [ocm_cmnd, "get", "/api/clusters_mgmt/v1/clusters"]
```

Comment:
You could use this endpoint [/api/clusters_mgmt/v1/clusters/{cluster_id}](https://api.stage.openshift.com/#/default/get_api_clusters_mgmt_v1_clusters__cluster_id_) to only get the cluster you want instead of going through all the clusters to find the right one.

Reply:
Its a good point, but at this point, I dont know the cluster_id, as you can see on line 101, the mgmt_cluster string could point to the ID or the cluster name, so I need to get all the clusters and find mine